src/lib/tls/asio/asio_stream_core.h

```diff
@@ -0,0 +1,120 @@
+/*
+* TLS Stream Helper
+* (C) 2018-2019 Jack Lloyd
+*     2018-2019 Hannes Rantzsch, Tim Oesterreich, Rene Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_ASIO_STREAM_CORE_H_
+#define BOTAN_ASIO_STREAM_CORE_H_
+
+#if defined(BOTAN_HAS_TLS) && defined(BOTAN_HAS_BOOST_ASIO)
+
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106600
+
+#include <boost/beast/core/flat_buffer.hpp>
+#include <botan/internal/asio_includes.h>
+#include <botan/tls_callbacks.h>
+#include <mutex>
+#include <vector>
+
+namespace Botan {
+
+namespace TLS {
+
+/**
+ * Contains the buffers for reading/sending, and the needed botan callbacks
+ */
+struct StreamCore : public Botan::TLS::Callbacks
+   {
+      StreamCore()
+         : m_input_buffer_space(17 * 1024, '\0'), // enough for a TLS Datagram
+           input_buffer(m_input_buffer_space.data(), m_input_buffer_space.size()) {}
+
+      virtual ~StreamCore() = default;
+
+      void tls_emit_data(const uint8_t data[], std::size_t size) override
+         {
+         m_send_buffer.commit(
```

Comment:
as far as I understand `tls_emit_data()` is (naïvely) meant to write the data onto a socket of some sort. It makes sense that we just prepare the send buffer here, but probably we should explain ourselves with a short comment. Especially, that this implementation assumes some other entity to actually go ahead and perform the actual "send".